src/test/scala/org/broadinstitute/dsde/firecloud/webservice/TrialApiServiceSpec.scala

```diff
@@ -97,6 +98,17 @@ final class TrialApiServiceSpec extends BaseServiceSpec with UserApiService with
       .respond(
         org.mockserver.model.HttpResponse.response()
           .withHeaders(MockUtils.header).withStatusCode(InternalServerError.intValue))
+
+    // For testing Thurloe returning no KVPs
+    localThurloeServer
+      .when(request()
+        .withMethod("GET")
+        .withHeader(fireCloudHeader.name, fireCloudHeader.value)
+        .withPath(UserApiService.remoteGetAllPath.format(userWithNoKVPs)))
+      .respond(
+        org.mockserver.model.HttpResponse.response()
+          .withHeaders(MockUtils.header).withStatusCode(OK.intValue)
+          .withBody(profile(userWithNoKVPs, Map.empty[String, String]).toJson.compactPrint))
```

Comment:
@davidangb Could you please double-check if I got this mocking right? I wanted to mimic the case where a user has no KVPs (inspired by the discussion on @ahaessly's PR#580). See also the tests added accordingly (look for `userWithNoKVPs`).

Reply:
This is functionally correct (footnote below), but I think it could be coded differently.

In the `allUsersAndProps` at line 70, `registeredUser` maps to `noTrialProps`, and `noTrialProps` is already a `Map.empty[String,String]` ... so you don't need this new "// For testing Thurloe returning no KVPs" mock, and you don't need the `userWithNoKVPs`, unless I'm missing something?

footnote: for a registered user who has never been enabled or disabled, Thurloe won't actually return an empty map - it will return KVPs that populate the user's `Profile`: see https://github.com/broadinstitute/firecloud-orchestration/blob/develop/src/main/scala/org/broadinstitute/dsde/firecloud/model/Profile.scala#L82. But it won't return any keys that would populate any part of `UserTrialStatus`. So the tests in here are valid, but to make them the most real, you could append a set of `Profile` KVPs to each response. I don't think you need to do this in order to merge this PR - the comment is for your edification.